src/components/DistanceEReceipt.js

```diff
@@ -0,0 +1,121 @@
+import * as ReportUtils from '../libs/ReportUtils';
+import * as CurrencyUtils from '../libs/CurrencyUtils';
+import * as TransactionUtils from '../libs/TransactionUtils';
+import tryResolveUrlFromApiRoot from '../libs/tryResolveUrlFromApiRoot';
+import ThumbnailImage from './ThumbnailImage';
+import useLocalize from '../hooks/useLocalize';
+import Icon from './Icon';
+import themeColors from '../styles/themes/default';
+import * as Expensicons from './Icon/Expensicons';
+import EReceiptBackground from '../../assets/images/eReceipt_background.svg';
+import useNetwork from '../hooks/useNetwork';
+import PendingMapView from './MapView/PendingMapView';
+
+const propTypes = {
+    /** The transaction for the distance request */
+    transaction: transactionPropTypes,
+};
+
+const defaultProps = {
+    transaction: {},
+};
+
+function DistanceEReceipt({transaction}) {
+    const {translate} = useLocalize();
+    const {isOffline} = useNetwork();
+    const {thumbnail} = TransactionUtils.hasReceipt(transaction) ? ReceiptUtils.getThumbnailAndImageURIs(transaction.receipt.source, transaction.filename) : {};
+    const {amount: transactionAmount, currency: transactionCurrency, merchant: transactionMerchant, created: transactionDate} = ReportUtils.getTransactionDetails(transaction);
+    const formattedTransactionAmount = transactionAmount ? CurrencyUtils.convertToDisplayString(transactionAmount, transactionCurrency) : translate('common.tbd');
+    const thumbnailSource = tryResolveUrlFromApiRoot(thumbnail || '');
+    const waypoints = lodashGet(transaction, 'comment.waypoints', {});
+    const sortedWaypoints = useMemo(
+        () =>
+            // The waypoint keys are sometimes out of order
+            _.chain(waypoints)
+                .keys()
+                .sort((keyA, keyB) => TransactionUtils.getWaypointIndex(keyA) - TransactionUtils.getWaypointIndex(keyB))
+                .map((key) => ({[key]: waypoints[key]}))
+                .reduce((result, obj) => (obj ? _.assign(result, obj) : result), {})
+                .value(),
+        [waypoints],
+    );
+    return (
+        <View style={[styles.flex1, styles.alignItemsCenter]}>
+            <ScrollView
+                style={styles.w100}
+                contentContainerStyle={[styles.flexGrow1, styles.justifyContentCenter, styles.alignItemsCenter]}
+            >
+                <View style={styles.eReceiptPanel}>
+                    <EReceiptBackground
+                        style={styles.eReceiptBackground}
+                        pointerEvents="none"
+                    />
+                    <View style={[styles.moneyRequestViewImage, styles.mh0, styles.mt0, styles.mb5, styles.borderNone]}>
+                        {isOffline || !thumbnailSource ? (
+                            <PendingMapView />
+                        ) : (
+                            <ThumbnailImage
+                                previewSourceURL={thumbnailSource}
```

Comment:
Hi, we had a bug here - App crashes when tapping on the receipt thumbnail while receipt is generating

On native, the pending attachment uses a static image, and it has the source: 1 (that's a standard RN, see [this](https://stackoverflow.com/questions/52398714/react-native-requireimage-returns-number)), which is not a string so it breaks the Str.isPDF() and Str.isImage() checks.

More details: https://github.com/Expensify/App/issues/28814